src/Nethermind/Nethermind.Network/P2P/Subprotocols/Les/LesProtocolHandler.cs

```diff
@@ -0,0 +1,209 @@
+                BestHash = head.Hash,
+                HeadBlockNo = head.Number,
+                GenesisHash = SyncServer.Genesis.Hash,
+
+                // TODO - implement config option for these
+                ServeHeaders = true,
+                ServeChainSince = 0x00,
+                //if (config.recentchain != null)
+                //    ServeRecentChain = Config.recentchain
+                ServeStateSince = 0x00,
+                //if (Config.serverecentstate != null)
+                //    ServeRecentState = Config.RecentState
+                TxRelay = true,
+                // TODO - should allow setting to infinite
+                BufferLimit = int.MaxValue,
+                MaximumRechargeRate = int.MaxValue
+            };
+            Send(statusMessage);
+
+            if (NetworkDiagTracer.IsEnabled) NetworkDiagTracer.ReportOutgoingMessage(Session.Node.Host, Name, statusMessage.ToString());
+            Metrics.LesStatusesSent++;
+
+            CheckProtocolInitTimeout().ContinueWith(x =>
+            {
+                if (x.IsFaulted && Logger.IsError)
+                {
+                    Logger.Error("Error during lesProtocol handler timeout logic", x.Exception);
+                }
+            });
+        }
+
+
+        public override byte ProtocolVersion { get; protected set; } = 3;
+
+        public override string ProtocolCode => Protocol.Les;
+
+        public override int MessageIdSpaceSize => 8;
+
+        protected override TimeSpan InitTimeout => Timeouts.Les3Status;
+
+        public byte RequestedAnnounceType = 0;
+
+        public override event EventHandler<ProtocolInitializedEventArgs> ProtocolInitialized;
+        public override event EventHandler<ProtocolEventArgs> SubprotocolRequested
+         {
+            add { }
+            remove { }
+        }
+
+        public override void AddSupportedCapability(Capability capability) { }
+
+        bool _statusReceived;
+        public override void HandleMessage(ZeroPacket message)
+        {
+            if (message.PacketType != LesMessageCode.Status && !_statusReceived)
+            {
+                throw new SubprotocolException($"No {nameof(StatusMessage)} received prior to communication with {Session.Node:c}.");
+            }
+
+            int size = message.Content.ReadableBytes;
+
+            switch (message.PacketType)
+            {
+                case LesMessageCode.Status:
+                    StatusMessage statusMessage = Deserialize<StatusMessage>(message.Content);
+                    if (NetworkDiagTracer.IsEnabled) NetworkDiagTracer.ReportIncomingMessage(Session.Node.Host, Name, statusMessage.ToString());
+                    Handle(statusMessage);
+                    break;
+            }
+        }
+
+        public void Handle(StatusMessage status)
+        {
+            Metrics.LesStatusesReceived++;
+
+            // set defaults
+            if (!status.AnnounceType.HasValue) status.AnnounceType = 1;
```

Comment:
shall that be set on the StatusMessage class instead?

Reply:
I assume you mean something to the effect of `public AnnounceType { get; set; } = 1;` in the definition?

The reason it's not currently is because `AnnounceType` can only be sent from the client. So, by having it actually default to `null`, it will default to not be sent, but when receiving a message we can still set it to the default if a client didn't specify.

Extra keys are required to be ignored, so according to the spec, sending it to a client should have no effect, but could be confusing at least